Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1089,3 +1090,259 @@ following aspects:
 - Security Flow description applied (String)
 - Endpoint token URL
 - URL to endpoint authorization ( If flow is based on "`authorizationCode`").	
+
+## 12. Subscription, Notification & Event
+
+In order to provide event-based interaction, CAMARA API could provided capabilities for subscription & notification management.
+A subscription allows an API consumer to request event notification reception at a given URL (callback address) for a specific context.
+A notification is the publication at the listener address about an occurred event.
+Managed event types are explicitly defined in CAMARA API OAS.
+
+### 12.1 Subscription
+
+We distinguish 2 types of subscriptions:
+- Instance-based subscription (indirect creation)
+- Resource-based subscription (direct creation)
+
+**Instance-based subscription**
+
+An instance-based subscription is a subscription indirectly created, additionally to another resource creation. For example for a Payment request (in Carrier Billing API), in the `POST/payments`, the API consumer could request to get event notification about **this** Payment request processing update. The subscription is not an autonomous entity and its lifecycle is linked to the managed entity (the Payment resource in this case). The subscription terminates with the managed entity.
+
+Providing this capability is optional for any CAMARA API depending on UC requirements.
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string | https callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentication token for callback API endpoint. It MUST be indicated within HTTP Authorization header accordingly to OAuth 2.0 Bearer Token format, e.g. ```Authorization: Bearer $notificationAuthToken``` | optional |
+
+Format conventions regarding ```notificationAuthToken``` attribute, in order to provide Uniqueness, Randomness and Simplicity for its management are the following:
```

Comment:
Lines 1120-1124 can be removed in my opinion as `notificationAuthToken` is now explicitly stated as an OAuth2 token. I am not sure why we need to add further explanation or restrictions on an OAuth2 token.

Reply:
@PedroDiez Ok for you as you added these lines?

Reply:
@PedroDiez - any updates on this please? 
@bigludo7 - I am OK with all the other updates. Thanks.

Reply:
Going to review updates today and i comment

Reply:
@PedroDiez - Thanks. I am not sure about the justification on keeping these 4 points restricting the OAuth2 token properties. For e.g. Why it MUST be an opaque OAuth2 token (or) MUST NOT be JWT (or) the length restriction?

Reply:
I am proposing some refinement.

The fact of indicating `notificationAuthToken` as an OAuth2 token makes sense as we are refering to it as a "Bearer" Token.
At the end how the API consumer generates it is out of the scope of Telco Provider.

With term "Opaque" means that the token used by API consumer should not be based in some info shared between API Consumer and Telco Operator, in order to not compromise security

Regarding size, idea below is to not allow API consumers to input long size tokens as they have to be stored and maintained by Telco Operator. Have observed that size increase has been agreed in last commonalities meeting, and it is OK, 256 length is length enough.

Reply:
Thanks. As commented [here](https://github.com/camaraproject/WorkingGroups/pull/173#issuecomment-1541914398), marking this conversation as resolved.